cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/SparkUtils.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.spark;
+
+import com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+
+/**
+ * A utility class to help determine Spark supports and locating Spark jar.
+ */
+public final class SparkUtils {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SparkUtils.class);
+
+  // Environment variable name for locating spark assembly jar file
+  private static final String SPARK_ASSEMBLY_JAR = "SPARK_ASSEMBLY_JAR";
+  // Environment variable name for locating spark home directory
+  private static final String SPARK_HOME = "SPARK_HOME";
+
+  private static File sparkAssemblyJar;
+
+  /**
+   * Locates the spark-assembly jar from the local file system.
+   *
+   * @return the spark-assembly jar location
+   * @throws IllegalStateException if cannot locate the spark assembly jar
+   */
+  public static synchronized File locateSparkAssemblyJar() {
+    if (sparkAssemblyJar != null) {
+      return sparkAssemblyJar;
+    }
+
+    // If someone explicitly set the location, use it.
+    // It's useful for overridding what being set for SPARK_HOME
+    String jarEnv = System.getenv(SPARK_ASSEMBLY_JAR);
+    if (jarEnv != null) {
+      File file = new File(jarEnv);
+      if (file.isFile()) {
+        LOG.info("Located Spark Assembly JAR in {}", file);
+        sparkAssemblyJar = file;
+        return file;
+      }
+    }
```

Comment:
The current logic doesn't check file extension nor the file content to see if it s a JAR or not. Maybe we should check the file extension? Not sure if worth it.